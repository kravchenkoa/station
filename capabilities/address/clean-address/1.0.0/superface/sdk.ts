import { createTypedClient } from "@superfaceai/one-sdk";

import { addressCleanAddress } from "./types/address/clean-address";

export { AddressCleanAddressProfile } from "./types/address/clean-address";
export const typeDefinitions = {
    ...addressCleanAddress
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
