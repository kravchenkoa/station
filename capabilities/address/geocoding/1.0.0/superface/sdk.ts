import { createTypedClient } from "@superfaceai/one-sdk";

import { addressGeocoding } from "./types/address/geocoding";

export { AddressGeocodingProfile } from "./types/address/geocoding";
export const typeDefinitions = {
    ...addressGeocoding
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
