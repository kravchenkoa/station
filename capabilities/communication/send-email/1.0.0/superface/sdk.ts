import { createTypedClient } from "@superfaceai/one-sdk";

import { communicationSendEmail } from "./types/communication/send-email";

export { CommunicationSendEmailProfile } from "./types/communication/send-email";
export const typeDefinitions = {
    ...communicationSendEmail
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
