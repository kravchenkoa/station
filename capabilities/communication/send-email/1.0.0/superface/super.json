{
  "profiles": {
    "communication/send-email": {
      "file": "./grid/profile.supr",
      "providers": {
        "sendgrid": {
          "file": "./grid/maps/sendgrid.suma"
        }
      }
    }
  },
  "providers": {
    "sendgrid": {
      "file": "../../../../../providers/sendgrid.json",
      "security": []
    }
  }
}