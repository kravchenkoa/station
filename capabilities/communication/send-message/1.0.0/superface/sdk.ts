import { createTypedClient } from "@superfaceai/one-sdk";

import { communicationSendMessage } from "./types/communication/send-message";

export { CommunicationSendMessageProfile } from "./types/communication/send-message";
export const typeDefinitions = {
    ...communicationSendMessage
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
