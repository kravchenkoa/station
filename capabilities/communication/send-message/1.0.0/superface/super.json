{
  "profiles": {
    "communication/send-message": {
      "file": "./grid/profile.supr",
      "providers": {
        "plivo": {
          "file": "./grid/maps/plivo.suma"
        },
        "twilio": {
          "file": "./grid/maps/twilio.suma"
        },
        "tyntec": {
          "file": "./grid/maps/tyntec.suma"
        },
        "vonage-nexmo": {
          "file": "./grid/maps/vonage-nexmo.suma"
        }
      }
    }
  },
  "providers": {
    "plivo": {
      "file": "../../../../../providers/plivo.json",
      "security": []
    },
    "twilio": {
      "file": "../../../../../providers/twilio.json",
      "security": []
    },
    "tyntec": {
      "file": "../../../../../providers/tyntec.json",
      "security": []
    },
    "vonage-nexmo": {
      "file": "../../../../../providers/vonage-nexmo.json",
      "security": []
    }
  }
}