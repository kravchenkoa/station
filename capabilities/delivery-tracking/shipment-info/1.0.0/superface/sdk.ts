import { createTypedClient } from '@superfaceai/one-sdk';

import { deliveryTrackingShipmentInfo } from './types/delivery-tracking/shipment-info';

export { DeliveryTrackingShipmentInfoProfile } from "./types/delivery-tracking/shipment-info";
export const typeDefinitions = {
    ...deliveryTrackingShipmentInfo
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
