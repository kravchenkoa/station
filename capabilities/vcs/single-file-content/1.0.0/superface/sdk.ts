import { createTypedClient } from '@superfaceai/one-sdk';

import { vcsSingleFileContent } from './types/vcs/single-file-content';

export { VcsSingleFileContentProfile } from "./types/vcs/single-file-content";
export const typeDefinitions = {
    ...vcsSingleFileContent
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
