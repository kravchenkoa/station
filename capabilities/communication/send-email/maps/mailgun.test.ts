import { SuperfaceClient } from '../../../../superface/sdk';

describe('communication/send-email/mailgun', () => {
  it('performs correctly', async () => {
    const client = new SuperfaceClient();
    const profile = await client.getProfile('communication/send-email');
    const useCase = profile.getUseCase('SendEmail');
    const provider = await client.getProvider('mailgun');

    expect(useCase).not.toBeUndefined();
    expect(provider).not.toBeUndefined();

    const result = await useCase.perform(
      {
        from: process.env.MAILGUN_COMMUNICATION_SENDEMAIL_FROM,
        to: process.env.COMMUNICATION_SENDEMAIL_TO,
        subject: 'Station test',
        text: 'Station test',
      },
      { provider }
    );

    expect(typeof result.unwrap().messageId).toEqual('string');
  });
});
