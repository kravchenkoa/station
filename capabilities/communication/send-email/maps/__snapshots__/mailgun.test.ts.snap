// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`communication/send-email/mailgun SendEmail when all inputs are correct should return messageId as result 1`] = `
Object {
  "value": Object {
    "messageId": "<20210924075951.1.7AF8B5427F9D5288@demo.superface.org>",
  },
}
`;

exports[`communication/send-email/mailgun SendEmail when inputs are invalid should throw an exception 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[0].statements[4].responseHandlers[1].statements[0]
Original Map Location: Line 38, column 7",
}
`;
